stdlib/public/core/AutoDiff.swift

```diff
@@ -170,6 +170,39 @@ public func differentiableFunction<T, U, R>(
   return original
 }
 
+public extension Differentiable {
+  @differentiable(wrt: self, vjp: _vjpWithGrad)
+  func withGradient(_ body: @escaping (inout CotangentVector) -> Void) -> Self {
```

Comment:
`withGradient(_ body: @escaping (CotangentVector) -> CotangentVector)` seems like a more natural API at first glance. Do you have a reason to do it this way?

Reply:
Discussed in person

Reply:
I think my core rationale is that `withXXX` APIs in Swift usually return whatever the closure returns, e.g. `withUnsafePointer(_:)` and `withoutAcutallyEscaping(_:)`, so `x.withGradient { $0 + 1 }` would syntactically indicate to a Swift programmer that this is "returning the gradient plus one in the forward pass".